### ISSUE TITLE
vlang 0.2.2 is broken

### ISSUE THREAD

**Issue by chekoopa** [NixOS/nixpkgs#131539]
### Describe the bug
V toolchain (incl. REPL, new project, doctor, module management) doesn't work properly being installed from Nix, as tools folder is write-protected.

### Steps To Reproduce
1. get nixpkgs at d8f8f31af9d77a48220e4e8a301d1e79774cb7d2
2. get `vlang` from these nixpkgs
3. try running `v`

### Expected behavior
V utils such as REPL run properly.

### Screenshots
```bash
[nix-shell:~]$ LANG= v
Welcome to the V REPL (for help with V itself, type `exit`, then run `v help`).
failed    cmd: "/nix/store/9hlj0981a37a5npiy2ylqjs9y1fs524n-vlang-weekly.2021.25/lib/v" "/nix/store/9hlj0981a37a5npiy2ylqjs9y1fs524n-vlang-weekly.2021.25/lib/cmd/tools/vrepl.v"
failed   code: 1
V panic: builder error: folder `/nix/store/9hlj0981a37a5npiy2ylqjs9y1fs524n-vlang-weekly.2021.25/lib/cmd/tools` is not writable

v hash: 19dca02
addr2line: 'v': No such file
v() [0x5c65a7]
addr2line: 'v': No such file
v() [0x512e8e]
addr2line: 'v': No such file
v() [0x40256e]
                                                        | 0x7f8465ab2ded | /nix/store/wvgyhnd3rn6dhxzbr5r71gx2q9mhgshj-glibc-2.32-48/lib/libc.so.6(__libc_start_main+0xed) 
addr2line: 'v': No such file
v() [0x4025aa]

[nix-shell:~]$ LANG= v new
failed    cmd: "/nix/store/9hlj0981a37a5npiy2ylqjs9y1fs524n-vlang-weekly.2021.25/lib/v" "/nix/store/9hlj0981a37a5npiy2ylqjs9y1fs524n-vlang-weekly.2021.25/lib/cmd/tools/vcreate.v"
failed   code: 1
V panic: builder error: folder `/nix/store/9hlj0981a37a5npiy2ylqjs9y1fs524n-vlang-weekly.2021.25/lib/cmd/tools` is not writable

<!-- same output down here-->
```

### Additional context
The possible solution is to pre-compile utils somehow (but there are many of them). 

### Notify maintainers

@chiiruno @nagy 

### Metadata
```console
[user@ckopo-nix:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.122, NixOS, 20.09.4346.d6a5be86b6a (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.11`
 - channels(root): `"home-manager-20.09, nixos-20.09.4346.d6a5be86b6a"`
 - channels(user): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: 
# a list of nixos modules affected by the problem
module:
```

**Comment by chekoopa** [comment]
Seems the only proper solution is running it from Git source. Libraries are bound to shells and FHS, though.
https://github.com/vlang/v/issues/10449#issuecomment-860280627

**Comment by Artturin** [comment]
im going to try to fix this by compiling them during the build

**Comment by Artturin** [comment]
nearly done just waiting on this bug https://github.com/vlang/v/issues/10974

**Comment by nagy** [comment]
What I tested during my earlier vlang PR was the compilation of files, which still seem to work.
Maybe we should add a checkPhase to the derivation like so:

```nix
doCheck = true;
checkPhase = ''
  runHook preCheck
  echo 'println("hello world")' > hello.v
  $out/bin/v hello.v
  ./hello
  runHook postCheck
'';
```

Of course we could/should also check the repl.

**Comment by Madouura** [comment]
Is this still relevant?

**Comment by nagy** [comment]
Yes, unfortunately. You can test it like this:

```
$ nix-shell -p vlang --run v
Welcome to the V REPL (for help with V itself, type  exit , then run  v help ).
  NB: the REPL is highly experimental. For best V experience, use a text editor,
  save your code in a  main.v  file and execute:  v run main.v
failed    cmd: "/nix/store/57vd0nqyzh3dd2xkxpvs8hymq871cqb0-vlang-weekly.2021.51/lib/v" -skip-unused "/nix/store/57vd0nqyzh3dd2xkxpvs8hymq871cqb0-vlang-weekly.2021.51/lib/cmd/tools/vrepl.v"
failed   code: 1
builder error: folder `/nix/store/57vd0nqyzh3dd2xkxpvs8hymq871cqb0-vlang-weekly.2021.51/lib/cmd/tools` is not writable
```

**Comment by buckley310** [comment]
I've written this module as a workaround so I can play with V while waiting for upstream.
https://github.com/buckley310/nixos-config/blob/cb8e891a3e24a5b1db9f24d8837f03c83f4b83b2/modules/vlang-fix.nix

**Comment by Madouura** [comment]
Fixed in  #173072
